packages/FrontC/FrontC.3.4.3/opam

```diff
@@ -0,0 +1,49 @@
+opam-version: "2.0"
+authors: "Hugues Cassé <casse@irit.fr>"
+homepage: "https://github.com/BinaryAnalysisPlatform/FrontC"
+bug-reports: "https://github.com/BinaryAnalysisPlatform/FrontC/issues"
+dev-repo: "git+https://github.com/BinaryAnalysisPlatform/FrontC"
+maintainer: "Ivan Gotovchits <ivg@ieee.org>"
+license: "LGPL-2.1-only"
+depends: [
+  "ocaml"
+]
+
+build: [
+  [make "PREFIX=%{lib}%" "OCAML_SITE=%{lib}%"]
+]
+
+install: [
+  [make "install" "PREFIX=%{lib}%" "OCAML_SITE=%{lib}%"]
+  ["cp" "META" "%{lib}%/FrontC"]
+]
+synopsis: "Library providing a C parser and lexer"
+description: """
+FrontC is an OCAML library providing a C parser and lexer. The result
+is a syntactic tree easy to process with usual OCAML tree management.
+
+It provides support for ANSI C syntax, old-C K&R style syntax and the
+standard GNU CC attributes.
+
+It provides also a C pretty printer as an example of use."""
+
+url {
+  src:
+    "https://github.com/BinaryAnalysisPlatform/FrontC/archive/refs/tags/v3.4.3.tar.gz"
+  checksum: [
+    "sha256=b96098bc56008e4239e571bb4e7407ac3b83be1c89bf945e2048063f9a5e528c"
+    "md5=399f66735541ecf8e06220618eef5c98"
+  ]
+  mirrors: "https://mirrors.aegis.cylab.cmu.edu/bap/FrontC/v3.4.3.tar.gz"
+}
+extra-source "FrontC.install" {
+  src:
+    "https://raw.githubusercontent.com/ocaml/opam-source-archives/main/patches/FrontC/FrontC.install"
+  checksum: [
+    "sha256=501dee92dc12ec57e4f34b5e5e15ee9630cb47c0c3f5219d6172a450c3495224"
+    "md5=c56e698d092d18179f9458f311c56412"
+  ]
+}
+x-opam-repository-commit-hash-at-time-of-archiving:
+  "2ac5b4411dc6433623d35bbb1ad092b393d3174e"
+x-reason-for-archiving: ["uninstallable"]
```

Comment:
```suggestion
x-reason-for-archiving: ["unavailable"]
```
uninstallable seems too vague. In this case, unavailable could mean `available: false` is it were the case here

Reply:
Thanks for your review. At some point (sorry, I was not around in the discussion), there was an agreement about which IDs to put into this reason-for-archiving, with the result https://github.com/ocaml/opam-repository/blob/master/governance/policies/archiving.md#specification-of-the-x--fields-used-in-the-archiving-process

So, `unavailable` is not a valid ID. We can of course revise these IDs, but that should be a separate process and not done within this PR.

Reply:
fair enough

Reply:
I think the idea, with uninstallable, was to mark packages that cannot be installed due to system constraints that can no longer be satisfied by any of the supported distributions (e.g. too recent gcc makes it uncompilable everywhere) as well as packages that are unavailable, the latter being recognizable by the fact that `available:false` is still there. I'd be happy to revise that.

Which brings me to my next point. I think we should not remove `available: false` since in a number of cases it was about packages that were voluntarily removed (for incompatibilities, security issues, ...). I would leave these packages uninstallable and only move them for "historical conservation" if somebody still wants to try them out by ignoring the flags

Reply:
I'm fine with both. Please discuss these two points at the meeting today where I won't be there.

I'm happy to have more IDs for the reason-for-removal. I'm as well fine to keep the `available: false` in this repository - if there's an agreement to both.

Reply:
I agree that "uninstallabe" seems more general (but actually less vague than "unavailable", IMO) and I think its semantics have been explained pretty clearly in the policy doc, and I also expected that packages that are `available: false` would be a special case of ones that `uninstallable`.

That said, I'm in favor of expanding or clarifying the policy, if that is called for. IMO, in this case, it seems fine to move forward here just by preserving `available: false`.

This latter point raises a general question: should any fields have their data altered during archiving?

According to the policy, under ["Archiving a Package"](https://github.com/ocaml/opam-repository/blob/master/governance/policies/archiving.md#archiving-a-package), the opam file is to be *extended* with the new `x-` fields, but it does not call for the alteration of any existing data. I would take this to imply that the package should otherwise be transferred without any alterations.

Reply:
On this topic, if archiving will require manual review, it occurs to me that it would be much easier to review archiving if it were possible to view only the diff of what fields have been added. Essentially we want a `mv` of each opam vile from the main repo to the archive, and the addition of two fields to record the provenance. Unfortunately I know of no obvious way to show the move between two different git repos.

However, if the archiving preceded thru two phases -- first commit opam file(s), then commit new `x-` fields -- we could at least see the additions clearly.

In this case, where I assume the archiving has been generated in bulk via an automated tool, such manual revue may not be needed tho.

Reply:
Ok, I'll adapt:
- preserve `available: false` in the opam files here
- the diff you're asking for is what I uploaded in a PR comment https://github.com/ocaml/opam-repository-archive/pull/2#issuecomment-2517204465

What are the changes? As you mention, adding the x- fields -- but also adjusting the existing "depends" section with upper bounds to whatever is the latest existing in the opam-repository.

The diff is generated by the archive-opam utility. I'd assume that at least 99% of archivals will be done by that tool.

Reply:
done in [98e2d4f](https://github.com/ocaml/opam-repository-archive/pull/2/commits/98e2d4f47ec883e763e787c115a858e9c9c58c8a) (leaves `available: false` in the opam files), also there's a new unified diff uploaded in https://github.com/ocaml/opam-repository-archive/pull/2#issuecomment-2520921225